### PR TITLE
feat(server-config): expose base path and api version as server config

### PR DESCRIPTION
**Pull request by machi1990**
This allows for dynamic configuration of these variables

Related to https://github.com/bf2fc6cc711aee1a0c2a/ffm-project/issues/8#issuecomment-1040724175

## Checklist (Definition of Done)

- [ ] Documentation added for the feature
- [ ] Code Review completed
- [ ] Verified independently by reviewer
- [ ] Equivalent Github Pull Request or an issue has been opened on the Quarkus based template

### CODE REVIEW

**Review by akoserwal — APPROVED** [review]
LGTM

**Review by miguelsorianod — APPROVED** [review]
Looks good to me.
If at some point we end up having multiple versions this would start to break down but it's not an issue at the moment

**Comment by machi1990** [comment]
> If at some point we end up having multiple versions this would start to break down but it's not an issue at the moment

True.

File: internal/dinosaur/internal/handlers/authentication.go [hunk 1]
```diff
@@ -32,9 +31,9 @@ func NewAuthenticationBuilder(ServerConfig *server.ServerConfig, KeycloakConfig
 			KeysURL(KeycloakConfig.OSDClusterIDPRealm.JwksEndpointURI). // sso SRE realm cert URL
 			Error(fmt.Sprint(errors.ErrorUnauthenticated)).
 			Service(errors.ERROR_CODE_PREFIX).
-			Public(fmt.Sprintf("^%s/%s/?$", routes.ApiEndpoint, routes.DinosaursFleetManagementApiPrefix)).
-			Public(fmt.Sprintf("^%s/%s/%s/?$", routes.ApiEndpoint, routes.DinosaursFleetManagementApiPrefix, routes.Version)).
-			Public(fmt.Sprintf("^%s/%s/%s/openapi/?$", routes.ApiEndpoint, routes.DinosaursFleetManagementApiPrefix, routes.Version)).
-			Public(fmt.Sprintf("^%s/%s/%s/errors/?[0-9]*", routes.ApiEndpoint, routes.DinosaursFleetManagementApiPrefix, routes.Version)),
+			Public(fmt.Sprintf("^%s/?$", ServerConfig.APIBasePath)).
+			Public(fmt.Sprintf("^%s/%s/?$", ServerConfig.APIBasePath, ServerConfig.APIVersion)).
+			Public(fmt.Sprintf("^%s/%s/openapi/?$", ServerConfig.APIBasePath, ServerConfig.APIVersion)).
```
Comment (emmanuelbernard) [inline internal/dinosaur/internal/handlers/authentication.go:36]: that raises an interesting question, when we have v2 of the API, what is implementing v1 so we support v1 and v2?

File: internal/dinosaur/internal/handlers/authentication.go [hunk 2]
```diff
@@ -32,9 +31,9 @@ func NewAuthenticationBuilder(ServerConfig *server.ServerConfig, KeycloakConfig
 			KeysURL(KeycloakConfig.OSDClusterIDPRealm.JwksEndpointURI). // sso SRE realm cert URL
 			Error(fmt.Sprint(errors.ErrorUnauthenticated)).
 			Service(errors.ERROR_CODE_PREFIX).
-			Public(fmt.Sprintf("^%s/%s/?$", routes.ApiEndpoint, routes.DinosaursFleetManagementApiPrefix)).
-			Public(fmt.Sprintf("^%s/%s/%s/?$", routes.ApiEndpoint, routes.DinosaursFleetManagementApiPrefix, routes.Version)).
-			Public(fmt.Sprintf("^%s/%s/%s/openapi/?$", routes.ApiEndpoint, routes.DinosaursFleetManagementApiPrefix, routes.Version)).
-			Public(fmt.Sprintf("^%s/%s/%s/errors/?[0-9]*", routes.ApiEndpoint, routes.DinosaursFleetManagementApiPrefix, routes.Version)),
+			Public(fmt.Sprintf("^%s/?$", ServerConfig.APIBasePath)).
```
[the remark this answers is not in the document]
Comment (machi1990) [reply internal/dinosaur/internal/handlers/authentication.go:34]: The config struct is defined at https://github.com/machi1990/ffm-fleet-manager-go-template/blob/a4df32806792f6be819b491d5a024dd13d857cc8/pkg/server/server_config.go#L8

The way it is populated is via CLI flags which are defined at: https://github.com/machi1990/ffm-fleet-manager-go-template/blob/a4df32806792f6be819b491d5a024dd13d857cc8/pkg/server/server_config.go#L40

and these flags (CLI options) are supplied to the pod via the service template: https://github.com/bf2fc6cc711aee1a0c2a/ffm-fleet-manager-go-template/blob/main/templates/service-template.yml#L661-L758

(the same could be done when the CLI is launched locally as a process or run as container)

File: internal/dinosaur/internal/handlers/authentication.go (hunk 1 above)
Comment (emmanuelbernard) [reply internal/dinosaur/internal/handlers/authentication.go:36]: If the state is all entirely shared from the same DB with no other caching and that the reconcile loops can be run concurrently, that works.

Comment (machi1990) [reply internal/dinosaur/internal/handlers/authentication.go:36]: > If the state is all entirely shared from the same DB with no other caching and that the reconcile loops can be run concurrently, that works.

Yes, thought it has its limits when DB changes. API versioning is something I'd wish we have an Epic for us to define our versioning strategy. I am happy to create one if we do not have it already @emmanuelbernard @davidffrench wdyt?

**Pull request merged**